src/Aether/Devices/Drivers/Sgp4x.cs

```diff
@@ -0,0 +1,210 @@
+            // 0xD4 = All tests passed
+            // 0x4B = One or more test failed
+            return testResultWithCRC[0] == 0xD4;
+        }
+
+        /// <summary>
+        /// Gets the raw VOC measurements from the device.
+        /// </summary>
+        /// <param name="relativeHumidityValue">The relative humidity value expressed as a percentage.</param>
+        /// <param name="temperatureValue">The temperature value expressed in degrees celsius.</param>
+        /// <returns>The raw VOC measurement value from the sensor. If an error occurred, it will be <see langword="null"/>.</returns>
+        /// <remarks>If default relative humidity and temperature values are supplied, humidity compensation will be disabled.</remarks>
+        public VolatileOrganicCompoundIndex? GetVOCRawMeasure(RelativeHumidity? relativeHumidityValue = null, Temperature? temperatureValue = null)
+        {
+            double rhValue = 50;
+            double tempValue = 25;
+
+            if(relativeHumidityValue is not null)
+            {
+                rhValue = relativeHumidityValue.Value.Value;
+
+                if (rhValue < 0 || rhValue > 100)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(relativeHumidityValue), relativeHumidityValue, "A relative humidity percentage value must be between 0 and 100");
+                }
+            }
+            
+            if (temperatureValue is not null)
+            {
+                tempValue = temperatureValue.Value.Value;
+
+                if (tempValue < -45 || tempValue > 130)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(temperatureValue), temperatureValue, "A temperature value must be between -45 and 130");
+                }
+            }
+            
+
+            Span<byte> writeBuffer = stackalloc byte[8];
+            Span<byte> readBuffer = stackalloc byte[3];
+
+            // Write start measure VOC raw command
+            writeBuffer[0] = 0x26;
+            writeBuffer[1] = 0x0F;
+
+            // Write default humidity value + CRC (0x80, 0x00, [CRC])
+            Sensirion.WriteUInt16BigEndianAndCRC8(writeBuffer.Slice(2, 3), (ushort)Math.Round(rhValue * 65535.00 / 100.00, MidpointRounding.AwayFromZero));
+
+            // Write default temperature value + CRC (0x66, 0x66, [CRC])
+            Sensirion.WriteUInt16BigEndianAndCRC8(writeBuffer.Slice(5, 3), (ushort)((tempValue + 45) * 65535.00 / 175.00));
+
+            // Transmit command
+            _device.Write(writeBuffer);
+
+            Thread.Sleep(30);
+
+            _device.Read(readBuffer);
+
+            // Read the results and validate CRC
+            ushort? readValue;
+            readValue = Sensirion.ReadUInt16BigEndianAndCRC8(readBuffer);
+
+            if (readValue is null)
+            {
+                throw new IOException("CRC for receved data is invalid");
```

Comment:
CRC failure is non-exceptional for the main read method.

```suggestion
                // CRC failed.
                return null;
```